server.py

```diff
@@ -173,19 +187,114 @@ def display_countries():
     return jsonify(country_information)
 
 
-@app.route('/signin')
-def sign_in():
-    pass
+@app.route('/login', methods=['GET'])
+def login_form():
+    """Show form for user login"""
+
+    return render_template("login_form.html")
+
+
+@app.route('/login', methods=['POST'])
+def login_process():
+    """Process login"""
+
+    email = request.form["email"]
+    password = request.form["password"]
+
+    user = User.query.filter_by(email=email).first()
+
+    if not user:
+        flash("No such user")
+        return redirect("/login")
+
+    if user.password != password:
+        flash("Incorrect password")
+        return redirect("/login")
+
+    session["user_id"] = user.user_id
+
+    flash("logged in")
+    return redirect(f"/")
+
+    # look at rating server.py after dealing with db to complete
+
+    # return redirect(f"/users/{user.user_id}")
+
+
+@app.route('/register', methods=['GET'])
+def register_form():
+    """show form for user signup"""
+
+    return render_template("register_form.html")
+
+
+@app.route('/register', methods=['POST'])
+def register_process():
+    """Process registration."""
+
+    first_name = request.form["first_name"]
+    last_name = request.form["last_name"]
+    email = request.form["email"]
+    password = request.form["password"]
+
+    new_user = User(fname=first_name, lname=last_name,
+                    email=email, password=password)
+
+    db.session.add(new_user)
+    db.session.commit()
+
+    flash(f"User {email} added.")
+    return redirect("/")
+
+
+@app.route('/logout')
+def logout():
+    """log out"""
+    del session["user_id"]
+    return redirect("/")
+
+
+@app.route('/user', methods=["GET"])
+def user_likes():
+    """display user's saved countries"""
+    current_user = session.get("user_id")
+    display_countries = Rating.query.filter_by(user_id=current_user).all()
+
+    countries_list = []
+
+    for country in display_countries:
+        countries_list.append(country.country_name)
+
+    return render_template("user_likes.html",
+                           countries=countries_list)
+
+
+@app.route('/user', methods=["POST"])
```

Comment:
Since the primary purpose of this route is to handle when a user likes a page, I would recommend renaming the end point. `/country-like` seems more appropriate. 

The reason being that `@app.route('/user', methods=["POST"])` is usually a high level route that's reserved for actions pertaining to creating or registering resources related to a `user`